### PR TITLE
Migrate downloading of saved bd to flask

### PR DESCRIPTION
**Pull request by cabarnes**
This addresses issue #39.

One thing that I noticed while doing this is that we have the `saved-bds` directory hardcoded a lot. I think we might want to either define it at the top or use a [flask configuration](http://flask.pocoo.org/docs/0.11/config/).

### CODE REVIEW

File: www/js/rover-api.js [hunk 1]
```diff
@@ -94,11 +93,23 @@ function acceptName() {
...
 }
+
+function downloadDesign(name) {
+  $.get(roverResource(['download', name]), function(response) {
+    var blob = new Blob([new XMLSerializer().serializeToString(response)]);
+    var downloadUrl = URL.createObjectURL(blob);
+    $(document.createElement('a'))
+      .attr( {'download': name, 'href': downloadUrl} )
+      .get(0)
+      .click();
```
Comment (aninternetof) [inline www/js/rover-api.js]: It's fascinating that this is the recommended way to do this. But, it doesn't actually end up looking that messy.

Comment (cabarnes) [inline www/js/rover-api.js]: Yeah, I'm hoping to get an opinion from @wavetylor about this. It seemed to be the way that other people did it, but seemed very odd.

Comment (wavetylor) [inline www/js/rover-api.js]: yea, that's exactly how we do it at work.

**Comment by wavetylor** [comment]
Looks good.

**Pull request merged**